test/cpp/end2end/async_end2end_test.cc

```diff
@@ -77,14 +77,17 @@ static int maybe_assert_non_blocking_poll(struct pollfd* pfds, nfds_t nfds,
 class PollOverride {
  public:
   PollOverride(grpc_poll_function_type f) {
-    prev_ = grpc_poll_function;
-    grpc_poll_function = f;
+    gpr_atm pf = reinterpret_cast<gpr_atm>(&grpc_poll_function);
+    prev_ = gpr_atm_full_xchg(&pf, reinterpret_cast<gpr_atm>(f));
```

Comment:
Ok, expressing this as reinterpret_cast's makes it much clearer, IMO. What is this supposed to be doing? You're exchanging into a local variable, right? Shouldn't this be something like

``` cpp
prev_ = gpr_atm_full_xchg(reinterpret_cast<gpr_atm*>(&grpc_poll_function), reinterpret_cast<gpr_atm>(f));
```

Reply:
My previous comment was on the code, but I still have the broader question about what the right approach should be.